src/pipx/animate.py

```diff
@@ -69,17 +69,22 @@ def print_animation(
     period: float,
     animate_at_beginning_of_line: bool,
 ):
-    (term_cols, _) = shutil.get_terminal_size(fallback=(9999, 24))
+    default_term_cols = 9999
+    (term_cols, _) = shutil.get_terminal_size(fallback=(default_term_cols, 24))
     event.wait(delay)
     while not event.wait(0):
         for s in symbols:
             if animate_at_beginning_of_line:
                 max_message_len = term_cols - len(f"{s} ... ")
+                if max_message_len <= 0:
+                    max_message_len = default_term_cols - len(f"{s} ... ")
```

Comment:
I would just do something like

```python
...
if max_message_len > 0:
    cur_line = f"{s} {message:.{max_message_len}}"
    if len(message) > max_message_len:
        cur_line += "..."
else:
    cur_line = f"{s} {message}"
```

since the fallback value is an arbitrary value anyway.

Reply:
In my mind I was wavering on whether the fallback `default_term_cols` should be effectively infinity (the way it is now) or a reasonable guess like 80 or 40.  So I wrote the code so that it would still work if we decided that `default_term_cols` should be some smaller value.

In general I couldn't think get this part of the code to look non-awkward to me, so if we want to enshrine the infinitely-long default line, I can change it to your suggestion.

Reply:
If we cannot determine a reasonable terminal width, do we want to guess a number like 80?  Or should we not truncate the message at all (current behavior)?

Reply:
Personally I would choose to not truncate, and let the terminal software (i.e. emulator these days) do what it wants. This generally results in either truncated or wrapped text, both are fine IMO.

Reply:
The wrapped text is the problem with animated lines, before we truncated, if the terminal was too narrow, the message would spill over to the next physical line.  Then the "delete line" terminal character would only go to the beginning of the last physical line wrapped extra bit, and not to the beginning of the entire multi-line.  This would make the animate thread spew animate frame lines down the terminal instead of rewriting the same line over and over as it was meant to.

Reply:
This conversation brings me back to an earlier thought.  I think we should just disable animation if we cannot determine the terminal width, or if it's uselessly-small (like < 10).  In such cases I would guess we are in some primitive situation anyway, or piping to something, etc.

@cs01, @uranusjr what do you think of this?

Reply:
> This conversation brings me back to an earlier thought. I think we should just disable animation if we cannot determine the terminal width, or if it's uselessly-small (like < 10). In such cases I would guess we are in some primitive situation anyway, or piping to something, etc.
> 
> @cs01, @uranusjr what do you think of this?

Yes I like this. I would favor keeping the code more simple+maintainable if the cost is occasionally not animating. Animating is really a "nice to have" kind of thing anyway. I don't think anyone will get too upset if apps are installed successfully but the animation didn't display.